os/arch/arm/src/imxrt/imxrt_mpuinit.c

```diff
@@ -118,18 +134,11 @@ void imxrt_mpu_initialize(void)
 #endif
 
 #ifdef CONFIG_BUILD_PROTECTED
-	/* Configure user flash and SRAM space */
-
-	DEBUGASSERT(USERSPACE->us_textend >= USERSPACE->us_textstart);
-
-	mpu_user_flash(USERSPACE->us_textstart, USERSPACE->us_textend - USERSPACE->us_textstart);
-
-	datastart = MIN(USERSPACE->us_datastart, USERSPACE->us_bssstart);
-	dataend = MAX(USERSPACE->us_dataend, USERSPACE->us_bssend);
-
-	DEBUGASSERT(dataend >= datastart);
+	int i;
 
-	mpu_user_intsram(datastart, dataend - datastart);
+	for (i = 0; i < (sizeof(regions_info) / sizeof(struct mpu_region_info)); i++) {
```

Comment:
How about: 
int i_max;
i_max = (sizeof(regions_info) / sizeof(struct mpu_region_info));

then use i_max in for loop. This will be increase overhead of a variable but more optimize as it  doesn't call sizeof function over an over again.

Reply:
this is resolved at compile time, I don't think it will have any overhead